Documentation/installation-with-spack.rst

```diff
@@ -0,0 +1,391 @@
+
+  # 4. with a gpu support
+  spack install seissol-env +mpi +asagi ^openmpi@3.1.5+cuda ^cuda@10.1.243 %gcc@8.3.0
+
+  # 5. with python, numpy and scipy
+  spack install seissol-env +mpi +asagi +python ^openmpi@3.1.5 %gcc@8.3.0
+
+
+Usage
+~~~~~
+
+.. code-block:: bash
+
+  module load seissol-env-develop-<compiler>-<hash>
+
+  # if you compile seissol-env with a compiler installed with Spack
+  # you may be required to load that compiler as well
+  module load <compiler>
+
+
+After that, you can compiler SiesSol using either CMake or 
+:ref:`Scons <compiling-seissol>`.
+
+
+SeisSol-Utils
+-------------
+
+By default, the scrip installs:
+
+- pumgen (without a Simmetrix support)
+- gmsh (without a GPU support)
+- gmsh2gambit
+- cube_c
+- rconv
+- SiesSol Cookbook, which contains some examples to run
+
+As in case of *seissol-env*, you need **scons** and, therefore, **python3** for 
+compiling. However, installation of these packages is optional to save 
+the set-up time.
+
+
+Options
+~~~~~~~
+
+- *benchmarks* [on, **default=off**] - installs SeisSol benchmarks. Make sure that you have access to the SeisSol LRZ-gitlab account.
+- *building_tools* [on, **default=off**] - installs scons and as a result python and pip
+- *gmsh_gui* [on, **default=off**] - enables gui support for gmsh
+- *paraview* [on, **default=off**] - installs Paraview for visualization
+
+Examples
+~~~~~~~~
+
+.. code-block:: bash
+
+  # 1. essential packages compiled with gcc compiler suite
+  spack install seissol-utils %gcc@8.3.0
+
+  # 2. with benchmarks and gmsh gmsh GUI
+  spack install seissol-utils+gmsh_gui+benchmarks %gcc@8.3.0
+
+  # 3. with gmsh GUI, paraview and scons
+  spack install seissol-utils+gmsh_gui+paraview+building_tools %gcc@8.3.0
+
+  # 4. essential packages with simmetrix support for pumgen
+  spack install seissol-utils ^pumgen+simmetrix_support %gcc@8.3.0 
+
+Usage
+~~~~~
+
+.. code-block:: bash
+
+  module load seissol-utils-develop-<compiler>-<hash>
+
+  # to access the Cookbook
+  cd $COOKBOOK
+
+  # to access the Benchmakrs
+  cd $BENCHMAKRS
+
+
+
+Tips and Tricks
+---------------
+
+.. _spack_known_issues:
+
+1. Spack builds the entire dependency graph before compaling and installing. 
+The graph includes all libs and packages which are necessary to build your 
+application, including packages like: *tar, gzip, zlib,  autoconf, 
+cmake, automake, pkgconf, m4, ncurses, etc*. Packages like these do not 
+affect performance of your application but help Spack to install it. 
+Therefore, it is not necessary to install them again and again. You can 
+install such  packages only once and mark them as Default 
+`(External) <https://spack-tutorial.readthedocs.io/en/latest/tutorial_configuration.html#external-packages>`_.
+and Non-Buildable. It can speed-up installation of SeisSol-Env and SeisSol-utils 
+considerably. You will need to modify and edit **~/.spack/packages.yaml** file.
```

Comment:
Wouldnt it be a good idea to reverse the process here? Make it the default to not build all these basic packages and tell people to also build tar, gzip, ... when they encounter problems.

Reply:
As far as I know there is only one way how to force spack to not build some packages: to add them to your ~/.spack/packages.yaml and mark them as Non-Buildable

Reply:
I would like to do the same as you suggested but I couldn't find a solution to enforce spack to prevent building these packages inside of an installation script

Reply:
Okay, so let's leave it like this.